### PR TITLE
Feature/eui 452 shuttering aa

### PR DESCRIPTION
**Pull request by adnanakgun**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-452

### Change description ###

Introduced HealthCheckGuard on the FE and a health check endpoint in node

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/lib/config/environments/demo.config.ts [hunk 1]
```diff
@@ -11,6 +11,18 @@ export default {
...
+        ccdDefApi: 'https://ccd-definition-store-api-aat.service.core-compute-aat.internal/health',
+        idamWeb: 'https://idam-web-public.aat.platform.hmcts.net/health',
+        idamApi: 'https://idam-api.aat.platform.hmcts.net/health',
+        s2s: 'https://rpe-service-auth-provider-demo.service.core-compute-demo.internal/health',
+        draftStoreApi: 'https://draft-store-service-demo.service.core-compute-demo.internal/health',
+        dmStoreApi: 'https://dm-store-demo.service.core-compute-demo.internal/health',
+        emAnnoApi: 'https://em-anno-demo.service.core-compute-demo.internal/health',
+        emNpaApi: 'https://em-npa-demo.service.core-compute-demo.internal/health',
+        rdProfessionalApi: 'https://rd-professional-api-preview.service.core-compute-preview.internal/health',
+    },
```
[the remark this answers is not in the document]
Comment (adnanakgun) [reply api/lib/config/environments/demo.config.ts]: I actually rebased on Vamshi's changes, if that's what you are asking.

File: src/shared/guards/health-check.guard.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+@Injectable()
+export class HealthCheckGuard implements CanActivate {
+    constructor(
+        private healthCheck: HealthCheckService,
+        private store: Store<fromRoot.State>,
+    ) {
+    }
+
+    canActivate() {
+        return this.checkStore().pipe(
```
Comment (fazanki) [inline src/shared/guards/health-check.guard.ts]: you re not checking the store so maybe you all this differently

File: src/app/app.routes.ts [hunk 3]
```diff
@@ -29,6 +30,10 @@ export const ROUTES: Routes = [
     canActivate: [AuthGuard],
     loadChildren: '../register/register.module#RegisterModule'
   },
+  {
+    path: 'service-down',
+    component: ShutterComponent
```
Comment (fazanki) [inline src/app/app.routes.ts]: not sure if service-down and StutterComponent should be called the same...

File: src/shared/guards/health-check.guard.ts (hunk 2 above)
Comment (adnanakgun) [reply src/shared/guards/health-check.guard.ts]: Changed function name

File: src/app/app.routes.ts (hunk 3 above)
Comment (adnanakgun) [reply src/app/app.routes.ts]: Changed component name

**Pull request merged**